src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -377,24 +456,37 @@ public boolean hasElevation() throws IOException {
 
                 // collect all unique values (can't do ranges now, we don't have a multi-attribute
                 // unique visitor)
-                UniqueVisitor visitor = new UniqueVisitor(attribute);
-                collection.accepts(visitor, null);
-                @SuppressWarnings("unchecked")
-                TreeSet<Object> result = new TreeSet<>(visitor.getUnique());
+                TreeSet<Object> result = null;
+                if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+                    result = getFixedValueRange(dimensionInfo.getFixedValues());
+                } else {
+                    UniqueVisitor visitor = new UniqueVisitor(attribute);
+                    collection.accepts(visitor, null);
+                    result = new TreeSet<>(visitor.getUnique());
+                }
                 return result;
             }
         }
 
         return null;
     }
 
-    /** Returns the max value for the elevation (as a Double, or as a NumberRange) */
+    /**
+     * Returns the max value for the elevation (as a Double, or as a NumberRange)
+     */
     public Double getMaxElevation() throws IOException {
         if (!hasElevation()) {
             return null;
         }
-        final String elevation =
-                reader.getMetadataValue(AbstractGridCoverage2DReader.ELEVATION_DOMAIN_MAXIMUM);
+        String elevation = null;
+        if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+            elevation =
+                    String.valueOf(getFixedValueRange(dimensionInfo.getFixedValues()));
```

Comment:
again why we split the array then we take back the TreeSet and convert it to a String?